controllers/common.go

```diff
@@ -57,6 +58,74 @@ func BuildCollector(params manifests.Params) ([]client.Object, error) {
 	return resources, nil
 }
 
+// @TODO: Later merge this in to reconcileDesiredObjects after implementing search functions for DCGM
+func reconcileDesiredObjectsWPrune(ctx context.Context, kubeClient client.Client, logger logr.Logger, owner v1alpha1.AmazonCloudWatchAgent, scheme *runtime.Scheme,
```

Comment:
nit: This is almost entirely the same function as `reconcileDesiredObjects`. The main difference is that it collects the "existing" `client.Object`s. Could we just make a helper function that does this for us and have the `reconcileDesiredObjects` function call that and ignore the slice?

```go
func helper(ctx context.Context, kubeClient client.Client, logger logr.Logger, owner metav1.Object, scheme *runtime.Scheme, desiredObjects ...client.Object) ([]client.Object, error)
```

Main concern is that if we pull in changes, we're likely only going to remember to update the `reconcileDesiredObjects` function, which is in upstream, and forget to update this one.